src/components/lib/cos_defcompinfo_api.c

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright 2016, Phani Gadepalli and Gabriel Parmer, GWU, gparmer@gwu.edu.
+ *
+ * This uses a two clause BSD License.
+ */
+
+#include <cos_defcompinfo_api.h>
+
+static struct cos_defcompinfo curr_defci;
+
+struct cos_defcompinfo *
+cos_defcompinfo_curr_get(void)
+{ return &curr_defci; }
```

Comment:
....but also the assertion, right?  I was not suggesting to only add a comment.  Please read my statement again.